src/net/sf/jsignpdf/SignerLogic.java

```diff
@@ -132,9 +133,24 @@ public boolean signFile() {
 		try {
 			SSLInitializer.init(options);
 
-			final PrivateKeyInfo pkInfo = KeyStoreUtils.getPkInfo(options);
-			final PrivateKey key = pkInfo.getKey();
-			final Certificate[] chain = pkInfo.getChain();
+			final PrivateKeyInfo pkInfo;
+			final PrivateKey key;
+			final Certificate[] chain;
+			// the 'cloudfoxy' crypto provider computes signatures externally and there are no
+			// certificates or keys available via Java CSPs -> they have to be pulled from an
+			// external source in 2 steps: 1. certificate chain, 2. signature itself
+			if (this.options.getKsType().toLowerCase().equals("cloudfoxy")){
```

Comment:
Could you extract the `"cloudfoxy"` to a constant?
Also the `getKsType()` may return `null`, so this would throw an NPE.